common/smoketest/key_vault_base_async.py

```diff
@@ -0,0 +1,19 @@
+from azure.identity.aio import DefaultAzureCredential
+from azure.identity import KnownAuthorities
+
+class KeyVaultBaseAsync:
```

Comment:
You can have KeyVaultBaseAsync inherit from KeyVaultBase and just overwrite `get_default_credential`

Or you could have just one class and allow it to take a credential, since that's the only thing that's different between these two classes